emcc.py

```diff
@@ -545,29 +545,16 @@ def get_all_js_syms():
   def build_symbol_list(filename):
     """Only called when there is no existing symbol list for a given content hash.
     """
-    library_syms = generate_js_symbols()
-    lines = []
+    library_syms = generate_js_sym_info()
 
-    for name, deps in library_syms.items():
-      if deps:
-        lines.append('%s: %s' % (name, ','.join(deps)))
-      else:
-        lines.append(name)
-    write_file(filename, '\n'.join(lines) + '\n')
+    write_file(filename, json.dumps(library_syms, separators=(',', ':')))
```

Comment:
This isn't quite as compact as the previous format, but JSON should be fast and it seems easier than maintaining our own serialize/deserialize now that there's more meta data.